build.cake

```diff
@@ -0,0 +1,183 @@
+var configuration = Argument("configuration", "Debug");
+
+//////////////////////////////////////////////////////////////////////
+// DISCOVERY VARS
+//////////////////////////////////////////////////////////////////////
+
+string[] SolutionList = null;
+string[] ProjList = null;
+var PROJ_EXT = "*.vbproj";
+
+//////////////////////////////////////////////////////////////////////
+// DEFINE RUN CONSTANTS
+//////////////////////////////////////////////////////////////////////
+
+var ROOT_DIR = Context.Environment.WorkingDirectory.FullPath;
+
+//////////////////////////////////////////////////////////////////////
+// ERROR LOG
+//////////////////////////////////////////////////////////////////////
+
+var ErrorDetail = new List<string>();
+
+//////////////////////////////////////////////////////////////////////
+// DISCOVER SOLUTIONS
+//////////////////////////////////////////////////////////////////////
+
+Task("DiscoverSolutions")
+.Does(() =>
+    {
+        SolutionList = System.IO.Directory.GetFiles(ROOT_DIR, "*.sln", SearchOption.AllDirectories);
+        ProjList = System.IO.Directory.GetFiles(ROOT_DIR, PROJ_EXT, SearchOption.AllDirectories);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// CLEAN
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+.IsDependentOn("DiscoverSolutions")
+.Does(() =>
+    {
+        foreach(var proj in ProjList)
+            CleanDirectory(DirFrom(proj) + "/bin/" + configuration);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// RESTORE PACKAGES
+//////////////////////////////////////////////////////////////////////
+
+Task("InitializeBuild")
+.IsDependentOn("DiscoverSolutions")
+.Does(() =>
+    {
+        foreach(var sln in SolutionList)
+            NuGetRestore(sln);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("Build")
+.IsDependentOn("InitializeBuild")
+.Does(() =>
+    {
+        foreach(var proj in ProjList)
+        {
+            var sample = System.IO.Path.GetFileNameWithoutExtension(proj);
+            DisplayHeading("Building " + sample + " sample");
+            BuildProject(proj, configuration);
+        }
+        
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEST
+//////////////////////////////////////////////////////////////////////
+
+Task("Test")
+.IsDependentOn("Rebuild")
+.OnError(exception => ErrorDetail.Add(exception.Message))
+.Does(() =>
+    {
+        foreach(var proj in ProjList)
+        {
+            var bin = DirFrom(proj) + "/bin/";
+            var projName = System.IO.Path.GetFileNameWithoutExtension(proj);
+            var dllName = bin + projName + ".dll";
+
+            DisplayHeading("Testing " + projName + " sample");
+
+            try
+            {
+                NUnit3(dllName);
```

Comment:
See discussion on nunit/nunit-csharp-samples#6 about pros and cons of using the NUnit3 tool.